rollouts/controllers/rollout_controller_test.go

```diff
@@ -0,0 +1,373 @@
+				(b) the number of attempts * interval period exceed the provided timeout value.
+				In the examples below, timeout and interval are Go Duration values of our choosing.
+			*/
+
+			rolloutLookupKey := types.NamespacedName{Name: RolloutName, Namespace: RolloutNamespace}
+			createdRollout := &gitopsv1alpha1.Rollout{}
+
+			// We'll need to retry getting this newly created Rollout, given that creation may not immediately happen.
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, rolloutLookupKey, createdRollout)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			Expect(createdRollout.Spec.Description).Should(Equal("Test Rollout"))
+
+			remoteSyncKey := types.NamespacedName{Name: "github-589324850-namespaces-e2e-sjc-0", Namespace: RolloutNamespace}
+			remoteSync := &gitopsv1alpha1.RemoteSync{}
+
+			// We should eventually have a remotesync object created corresponding to a target cluster
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, remoteSyncKey, remoteSync)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+			Expect(remoteSync.Spec.Template.Spec.Git.Repo).To(Equal("https://github.com/droot/store.git"))
+			Expect(remoteSync.Spec.Template.Spec.Git.Revision).To(Equal("v3"))
+
+			// We should eventually have the rollout completed
+			Eventually(func() bool {
+				_ = k8sClient.Get(ctx, rolloutLookupKey, createdRollout)
+				return createdRollout.Status.Overall == "Completed"
+			}, 1*time.Minute, interval).Should(BeTrue())
+
+			Expect(createdRollout.Status.ClusterStatuses).To(HaveLen(1))
+			Expect(createdRollout.Status.ClusterStatuses).To(ContainElement(gitopsv1alpha1.ClusterStatus{
+				Name: "e2e-sjc-0",
+				PackageStatus: gitopsv1alpha1.PackageStatus{
+					PackageID:  "github-589324850-namespaces-e2e-sjc-0",
+					Status:     "Synced",
+					SyncStatus: "Synced",
+				},
+			}))
+			forground := metav1.DeletePropagationForeground
+			Expect(k8sClient.Delete(context.TODO(), createdRollout, &client.DeleteOptions{PropagationPolicy: &forground})).To(Succeed())
+			// We should wait for the rollout to be deleted
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, rolloutLookupKey, createdRollout)
+				return client.IgnoreNotFound(err) == nil
+			}, 1*time.Minute, interval).Should(BeTrue())
+		})
+	})
+
+	Context("A progressive Rollout", func() {
+		// setup target clusters
+		BeforeEach(func() {
+			var err error
+			targets = []e2eclusters.Config{
+				{
+					Prefix: "e2e-sjcc-",
+					Count:  1,
+					Labels: map[string]string{
+						"city": "sjcc",
+					},
+				},
+				{
+					Prefix: "e2e-sfoo-",
+					Count:  1,
+					Labels: map[string]string{
+						"city": "sfoo",
+					},
+				},
+			}
+			targetClusterSetup, err = e2eclusters.GetClusterSetup(tt, k8sClient, targets...)
+			Expect(err).NotTo(HaveOccurred())
+
+			Expect(targetClusterSetup.PrepareAndWait(context.TODO(), 5*time.Minute)).To(Succeed())
```

Comment:
Are we expecting to use 5*time.Minute in every test? If so, maybe we can put it in a const

Reply:
So these timeouts really depend on the target cluster setup and that may vary between these tests, so avoiding the reuse.